### ISSUE TITLE
Add support for `.SSC` step files

### ISSUE THREAD

**Issue by AldoMX** [AldoMX/StepMania-AMX#2]
Too many players were requesting this feature in the past, and in theory it shouldn't be too troublesome to add support for it.

There are some points to take into consideration though:

1. The standard `.SSC` format is almost identical to a `.SM` file since they hadn't added new `NoteType`s, so it can be considered a subset of the `.SMA` format for the purposes of using the same call to `NoteDataUtil::LoadFromSMNoteDataString`.

2. The most exotic stuff available on the wild is achieved with `BGChange`s scripted in Lua in both: `.SM` (from v4) and `.SSC` formats, we should trigger a warning when the game is attempting to load a file with Lua stuff telling the player in a diplomatic tone `You know, this is fucking 3.9, use your common sense, a WinDEU step won't work here!`.

3. There are some pre-requisites we need to complete before attempting to load the first `.SSC` file, most notably, `WARP`s and `SCROLL`s in `TimingData`. I'll be opening an issue for them.

4. Unfortunately for everyone, there are 2 prominent projects that completely raped the standard `.SSC` format since the kids behind them just wanted to make a PIU replica and didn't care about the compatibility issues that they would create. [One of them](https://www.facebook.com/piuinfinity/) is a commercial game which encrypts the steps and it's a taboo to crack/decrypt/release it, so it shouldn't be a concern, but the [second one](http://stepf2.blogspot.mx/) now has hordes of people creating content for it. This means that there is an amount of extra work required in order to achieve a level of compatibility that won't end up in players whining. This means biting the bullet and hoping that this decision won't backfire by giving content creators the peace of mind of just releasing content in the non-standard `.SSC` format.

5. The 2 prominent projects I mentioned before somehow manged to add the same subset of features from PIU NX2 in their non-standard formats. I won't speculate how that subset of features ended up in both projects, but as a little life insurance I'll have to follow a [patented methodology from Microsoft](https://en.wikipedia.org/wiki/Embrace,_extend_and_extinguish), which basically means to add some low hanging fruit like a fake division mode (with STX support to load the all-grove and all-wild paths) and redesign `ScreenEdit` to make it easier to work with the new `NoteType`s with visibility and behavior attributes.
![](https://cloud.githubusercontent.com/assets/476531/23483204/9c2e1d1c-fe97-11e6-94b2-0a727137e1ad.jpg)
(Sorry for the mockup that looks like drawn by a kid, art is not one of my top skills)

6. Ok, this is where things get tricky... I mentioned that we should warn the player if they attempt to load a file with LUA `BGChange`s, but Prime began using stuff that players recreated with basic `FGChange`s... Should we add support for basic LUA stuff and warn when people try to access a binding that doesn't exist? Should we parse the LUA file with regex to find out whether it is something simple so we could rewrite it as an `.ini BGChange`? There's some food for thought here.
I really want to silently ignore the small BGChanges but warn the players when they load the crazy stuff from the OpenITG community, and I would rather let players use the packs of songs made for StepF2 than getting my hands dirty with proper conversions that don't follow the `n-2` rule.

In conclusion, while adding support for standard `.SSC` should be easy, it would be meaningless since people now create content in a non-standard `.SSC` format which requires playing catch-up and taking the lead again.
